### PR TITLE
fix xlsx report for diff

### PR DESCRIPTION
**Pull request by babenek**
## Description

Please include a summary of the change and which is fixed.

- Add deprecation for main scan function
- Fix Bug for xlsx of diff
- Modify report names of json for diff (py3.8 pathlib does not support stem manipulation)
- Update tests for diff report verification
- Use prefix ``.added`` instead name ends ``_added`` for JSON diff reports

## How has this been tested?

Please describe the tests that you ran to verify your changes.

- [x] UnitTest
- [x] Benchmark

### CODE REVIEW

File: credsweeper/__main__.py [hunk 1]
```diff
@@ -270,20 +271,30 @@ def get_json_filenames(json_filename: str):
...
+         xlsx_filename: Optional[str] = None) -> int:
     """Scan content_provider data, print results or save them to json_filename is not None
 
     Args:
         args: arguments of the application
         content_provider: FilesProvider instance to scan data from
-        json_filename: json type report file path or None
-        xlsx_filename: xlsx type report file path or None
+        json_filename: (Deprecated) json type report file path or None
+        xlsx_filename: (Deprecated) xlsx type report file path or None
```
Comment (csh519) [inline credsweeper/__main__.py:284]: It seems `json_filename` and `xlsx_filename` can be removed from `scan()`..
Could you please consider to remove those two parameters?

File: credsweeper/__main__.py [hunk 2]
```diff
@@ -270,20 +271,30 @@ def get_json_filenames(json_filename: str):
     return added_json_filename, deleted_json_filename
```
Comment (csh519) [inline credsweeper/__main__.py:271]: It seems there `get_json_filenames()` doesn't seem to be used anywhere..
Could you please remove this function?

**Review by csh519 — APPROVED** [review]
Thank you for fix the bug of xlsx file diff.

LGTM 👍

**Review by kmnls — APPROVED** [review]
Agree

**Pull request merged**